packages/search/src/search.tsx

```diff
@@ -90,349 +96,349 @@ type State = {
-			| MouseEvent< HTMLButtonElement | HTMLInputElement >
-			| KeyboardEvent< HTMLButtonElement | HTMLInputElement >
-	): void => {
-		event.preventDefault();
-
-		if ( this.props.disabled ) {
-			return;
-		}
-
-		this.setState( {
-			keyword: '',
-			isOpen: false,
-		} );
-
-		if ( this.searchInput.current ) {
-			this.searchInput.current.value = ''; // will not trigger onChange
-		}
-
-		if ( this.props.pinned ) {
-			this.searchInput.current?.blur();
-			this.openIcon.current?.focus();
-		} else {
-			this.searchInput.current?.focus();
-		}
-
-		this.props.onSearchClose?.( event );
-
-		this.props.recordEvent?.( 'Clicked Close Search' );
-	};
-
-	closeListener = keyListener( this.closeSearch );
-	openListener = keyListener( this.openSearch );
+//This is fix for IE11. Does not work on Edge.
+//On IE11 scrollLeft value for input is always 0.
+//We are calculating it manually using TextRange object.
+const getScrollLeft = (
+	inputElement: HTMLInputElement & { createTextRange?: () => Range }
+): number => {
+	//TextRange is IE11 specific so this checks if we are not on IE11.
+	if ( ! inputElement.createTextRange ) {
+		return inputElement.scrollLeft;
+	}
 
-	componentDidUpdate( _: Props, prevState: State ): void {
-		this.scrollOverlay();
-		// Focus if the search box was opened
-		if ( this.state.isOpen && ! prevState.isOpen ) {
-			this.focus();
-		}
+	const range = inputElement.createTextRange();
+	const inputStyle = window.getComputedStyle( inputElement, undefined );
+	const paddingLeft = parseFloat( inputStyle.paddingLeft );
+	const rangeRect = range.getBoundingClientRect();
+	const scrollLeft =
+		inputElement.getBoundingClientRect().left +
+		inputElement.clientLeft +
+		paddingLeft -
+		rangeRect.left;
+	return scrollLeft;
+};
 
-		if ( this.state.keyword === prevState.keyword ) {
-			return;
-		}
+type ImperativeHandle = {
+	focus: () => void;
+	blur: () => void;
+	clear: () => void;
+};
 
-		// if there's a keyword change: trigger search
-		if ( this.state.keyword ) {
-			this.onSearch( this.state.keyword );
+const InnerSearch: React.ForwardRefRenderFunction< ImperativeHandle, Props > = (
+	{
+		delaySearch: delaySearchProp,
+		disabled,
+		pinned,
+		onSearchClose,
+		onSearchChange,
+		onSearch: onSearchProp,
+		onBlur: onBlurProp,
+		onKeyDown: onKeyDownProp,
+		onClick,
+		describedBy,
+		delayTimeout,
+		defaultValue,
+		defaultIsOpen,
+		autoFocus,
+		onSearchOpen,
+		recordEvent,
+		overlayStyling,
+		placeholder: placeholderProp,
+		__,
+		inputLabel,
+		disableAutocorrect,
+		className,
+		dir,
+		fitsContainer,
+		searching,
+		compact,
+		hideOpenIcon,
+		openIconSide,
+		minLength,
+		maxLength,
+		hideClose,
+	},
+	forwardedRef
+) => {
+	const [ { keyword, isOpen, hasFocus }, setState ] = React.useReducer<
+		React.Reducer< State, Partial< State > >
+	>( ( oldState, nextState ) => ( { ...oldState, ...nextState } ), {
+		keyword: defaultValue ?? '',
+		isOpen: defaultIsOpen ?? false,
+		hasFocus: autoFocus ?? false,
+	} );
```

Comment:
What if we used three `useState` hooks here?
```js
const [ keyword, setKeyword ] = useState( defaultValue );
const [ isOpen, setOpen ] = useState( defaultIsOpen );
const [ hasFocus, setFocus ] = useState( autoFocus );
```
That's a more "hooky" coding style I would say -- the reducer and `setState` is a straightforward rewrite of the old-fashioned class component `setState`.